01/components/TableFooter.js

```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+
+class TableFooter extends React.Component {
+    render() {
+        const {data} = this.props
+        const totalSum = data.reduce( (total, element) => {
+            return element.price * element.quantity + total;
+        }, 0)
```

Comment:
Można by było też przekazać samą cenę do tego komponentu, a w głównym ją obliczyć.
Jednak należy traktować to rozwiązanie jak alternatywne, a nie lepsze ;)